### PR TITLE
Add --exclude argument to CLI

### PR DESCRIPTION
**Pull request by cblakkan**
* Adds `--exclude PATTERN` argument to the CLI
* Reorganizes the `-h` menu for better clarity
* Enforces all paths supplied to the command line interface must exist (will probably do a major version bump #171 )
* Supports `-n auto` to spawn child processes equal to number of system CPUs

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
     packages=find_packages(),
     include_package_data=True,
     install_requires=["pyyaml"],
-    extras_requires={"ruamel": ["ruamel.yaml"]},
+    extras_require={"ruamel": ["ruamel.yaml"]},
```
Comment (joecackler) [inline setup.py:21]: ha whoops

File: README.md [hunk 2]
```diff
@@ -50,24 +55,26 @@ looking up the directory tree until it finds one. If Yamale can not find a schem
 Usage:
 
 ```bash
```
Comment (joecackler) [inline README.md:57]: I don't think we want `bash` (or any) syntax hl here

File: yamale/command_line.py [hunk 3]
```diff
@@ -98,39 +100,82 @@ def _validate_dir(root, schema_name, cpus, parser, strict):
         raise ValueError("\n----\n".join(set(error_messages)))
 
 
-def _router(paths, schema_name, cpus, parser, strict=True):
+def _router(paths, schema_name, cpus, parser, excludes=None, strict=True, verbose=False):
+    EXCLUDE_REGEXES = tuple(re.compile(e) for e in excludes) if excludes else tuple()
+
+    def should_exclude(yaml_path):
+        should_exclude = any(pattern.search(yaml_path) for pattern in EXCLUDE_REGEXES)
```
Comment (joecackler) [inline yamale/command_line.py:107]: little hard to follow when `should_exclude` bool shadows `should_exclude` fn

File: yamale/command_line.py [hunk 4]
```diff
@@ -98,39 +100,82 @@ def _validate_dir(root, schema_name, cpus, parser, strict):
...
     )
-    parser.add_argument("-V", "--version", action="version", version=__version__)
     parser.add_argument("-s", "--schema", default="schema.yaml", help="filename of schema. Default is schema.yaml.")
-    parser.add_argument("-n", "--cpu-num", default=4, type=int, help="number of CPUs to use. Default is 4.")
+    parser.add_argument(
+        "-e",
+        "--exclude",
+        metavar="PATTERN",
+        action="append",
+        help="Python regex used to exclude files from validation. Any substring match of a files absolute path will be excluded. Uses deafult Python3 regex. Option can be supplied multiple times.",
```
Comment (joecackler) [inline yamale/command_line.py:145]: "of a files" => "of a file's"
"deafult" => "default"

File: yamale/command_line.py [hunk 5]
```diff
@@ -98,39 +100,82 @@ def _validate_dir(root, schema_name, cpus, parser, strict):
...
+        return int(num_cpu)
+
     parser = argparse.ArgumentParser(description="Validate yaml files.")
     parser.add_argument(
         "paths",
         metavar="PATH",
         default=["./"],
         nargs="*",
-        help="Paths to validate, either directories or files. Default is the current directory.",
+        help="paths to validate, either directories or files. Default is the current directory.",
```
Comment (joecackler) [inline yamale/command_line.py:137]: why lowercase?

Comment (cblakkan) [reply yamale/command_line.py:137]: I was just trying to make things consistent but will change it to be proper :)

**Review by joecackler — APPROVED** [review]

**Comment by cblakkan** [comment]
@joecackler I guess we'll release 6.0.0 due to the "breaking change behavior" of raising an error when a file isn't found.

**Comment by joecackler** [comment]
> @joecackler I guess we'll release 6.0.0 due to the "breaking change behavior" of raising an error when a file isn't found.

sgtm
